cloudify_agent/api/pm/base.py

```diff
@@ -235,8 +239,23 @@ def __init__(self, logger=None, **params):
         self.broker_ip = params['broker_ip']
         self.file_server_host = params['file_server_host']
 
-        # Optional parameters
+        # Optional parameters - REST client
         self.validate_optional()
+        self.rest_port = params.get(
+            'rest_port') or defaults.REST_PORT
+        self.rest_protocol = params.get(
+            'rest_protocol') or defaults.REST_PROTOCOL
+        self.verify_rest_certificate = params.get('verify_rest_certificate')
+        self.local_rest_cert_file = params.get('local_rest_cert_file', '')
+        self.rest_cert_content = params.get('rest_ssl_cert_content', '')
+        self.security_enabled = params.get('security_enabled')
+        # REST credentials need to be prefixed with _ so they're not stored
+        # when the daemon is serialized
+        self._rest_username = params.get('rest_username')
+        self._rest_password = params.get('rest_password')
+        self._rest_token = params.get('rest_token')
```

Comment:
all the REST parameters must be in place before `._get_name_from_manager` because that might attempt to use the REST client